games-service/src/catalogBatchProcess/catalogBatchProcess.ts

```diff
@@ -0,0 +1,85 @@
+import { Client } from "pg";
+import { SNS } from "aws-sdk";
+import "source-map-support/register";
+import { DB_OPTIONS } from "../db/dbConfig";
+import { REGION } from "../constants";
+
+export const catalogBatchProcess = async event => {
+  console.log("catalogBatchProcess was called with event: ", event);
+
+  const sns = new SNS({ region: REGION });
+  const client = new Client(DB_OPTIONS);
+  await client.connect();
+
+  try {
+    const products = event.Records.map(({ body }) => {
+      const product = JSON.parse(body);
+
+      if (!product.title) {
+        product.error = `Field 'title' is empty`;
+      }
+      return { ...product };
+    }).filter(p => {
+      if (p.error) {
+        console.error(`Product is not valid: ${JSON.parse(p)}`);
+      }
+
+      return !p.error;
+    });
```

Comment:
You could use some 3rd party library for validations. Like Joi, Yup etc. It's much easier and more scalable.

Reply:
Thanks, I know that I could ;)